rclc/CMakeLists.txt

```diff
@@ -33,6 +33,7 @@ find_package(rcl_action REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(rosidl_generator_c REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(Threads REQUIRED)
```

Comment:
@Acuadros95 currently, the Executor is single-threaded. Therefore Threads package is not necessary. However, I will merge the PR multi-threaded executor, so I will need this package. It's good to know, that we need this dependency.

Reply:
I added it because threads were being linked here:
https://github.com/ros2/rclc/blob/196d0f9bc831fc843fc7fe2b9f429058adc730bf/rclc/CMakeLists.txt#L82

Reply:
We need to add Thread dependency **only** in example repos.

We cannot add `find_package(Threads REQUIRED)` at `rclc` level (at least not unconditionally) because it will break the micro-ROS cross-compiled environments for thread-less platforms.

Reply:
resolved in https://github.com/ros2/rclc/pull/237/commits/da1cebeea58447ff216323994f11e649bdd1e3a4